### PR TITLE
chore(cli): extract vite config as a submodule to the library

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This PR extracts the duplicated vite plugins as `@storyblok/field-plugin/vite`.

## Why?

JIRA: EXT-1875

### CODE REVIEW

File: packages/cli/templates/js/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
+    "composite": true,
+    "module": "ESNext",
+    "moduleResolution": "bundler",
+    "resolveJsonModule": true,
+    "allowSyntheticDefaultImports": true
+  },
+  "include": ["vite.config.ts"]
+}
```
Comment (eunjae-lee) [inline packages/cli/templates/js/tsconfig.json:10]: Although we are not using TypeScript in the JS template, we need this file in order to `import { plugins } from '@storyblok/field-plugin/vite'` in the `vite.config.ts`.

File: packages/cli/templates/js/vite.config.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import { defineConfig, type PluginOption } from 'vite'
+import { defineConfig } from 'vite'
 import cssInjectedByJsPlugin from 'vite-plugin-css-injected-by-js'
-import * as querystring from 'querystring'
+import { plugins } from '@storyblok/field-plugin/vite'
```
Comment (eunjae-lee) [inline packages/cli/templates/js/vite.config.ts:3]: The submodule exports

- `plugins`
- `printProd`
- `printDev`

**Review by BibiSebi — APPROVED** [review]
Looking good, thank you for the explanations 🙏

**Comment by eunjae-lee** [comment]
@johannes-lindgren something we talked about. pinging you for visibility and in case you have another feedback

**Pull request merged**